data-integration/admin_units.py

```diff
@@ -110,12 +113,39 @@ def get(
     return admin_units
 
 
-if __name__ == "__main__":
-    logging.basicConfig(level=logging.INFO)
+def integrate():
+    """Integrate datasets."""
+    # Get spatial records
     admin_units = get(GISCO_DATASETS, crs=CRS.from_epsg(3035))
-    admin_units["ds_id"] = DS_ID
+    # Upload dataset record
     dataset = pd.DataFrame([{"ds_id": DS_ID}])
     utilities.toPostgreSQL(
         dataset, DB_URL, schema="datasets",
     )
+    # Upload spatial records
+    admin_units["ds_id"] = DS_ID
     utilities.toPostGIS(admin_units, DB_URL)
+    # Upload empty data records
+    data = admin_units.loc[:, ["ds_id", "fid"]].copy()
+    data["value"] = 0
+    data["variable"] = ""
+    utilities.toPostgreSQL(
+        data, DB_URL, schema="data",
+    )
+
+
+if __name__ == "__main__":
+    argv = sys.argv
+    if "--force" in argv:
+        isForced = True
+    else:
+        isForced = False
+    if utilities.datasetExists(DS_ID, DB_URL):
+        if isForced:
```

Comment:
Sorry I don't understand

Reply:
It's a similar remark as the one above the race condition, usually you try to bundle database operation that should be done in one operation into a single transaction. If you start running multiple pipeline in parallel or if something else is accessing the database, this will be a problem.